packages/next/build/webpack/require-hook.ts

```diff
@@ -0,0 +1,62 @@
+// sync injects a hook for webpack and webpack/... requires to use the internal ncc webpack version
+// this is in order for userland plugins to attach to the same webpack instance as next.js
+// the individual compiled modules are as defined for the compilation in bundles/webpack/packages/*
+
+const hookPropertyMap = new Map([
+  ['webpack', 'next/dist/compiled/webpack/webpack-lib'],
```

Comment:
These `next` paths need to be absolute otherwise you run into hoisting issues

Reply:
Should just be a matter of calling `require.resolve` on the properties, or even just a `path.join(__dirname, ...)`

Reply:
Fixing it in https://github.com/vercel/next.js/pull/21877